### PR TITLE
Add pannels/alerts for plank-job-with-infra-role

### PR DESCRIPTION
**Pull request by hongkailiu**
/cc @openshift/openshift-team-developer-productivity-test-platform

### CODE REVIEW

**Comment by hongkailiu** [comment]
preview on staging:
https://grafana-prow-monitoring-stage.svc.ci.openshift.org/d/8ce131e226b7fd2901c2fce45d4e21c1/dptp-dashboard?orgId=1&fullscreen&panelId=3

**Comment by hongkailiu** [comment]
Screenshot from 2019-10-07 17-27-58

File: cluster/ci/monitoring/mixins/grafana_dashboards/dptp.jsonnet [hunk 1]
```diff
@@ -39,4 +39,21 @@ dashboard.new(
     x: 0,
     y: 0,
   })
+.addPanel(
+    (graphPanel.new(
+        'Plank Infra-Jobs Failures',
+        description='sum(rate(prowjob_state_transitions{job="plank",job_name!~"rehearse.*",state="failure"}[5m])) by (job_namespace, job_name) * on (job_namespace, job_name) group_right prow_job_labels{job_agent="kubernetes",label_ci_openshift_io_role="infra"}',
```
Comment (stevekuznetsov) [inline cluster/ci/monitoring/mixins/grafana_dashboards/dptp.jsonnet]: I don't think we need to partition by namespace? We just want the number of jobs named `foo` that failed, right?

**Comment by stevekuznetsov** [comment]
Could we define a simple JSONNET function so that a user like @bparees could call 

```jsonnet
alert_on_job("my-job-name", "my-slack-channel")
```

In order to opt in a channel to job failure messages?

**Comment by bparees** [comment]
> Could we define a simple JSONNET function so that a user like @bparees could call

:pray:

File: cluster/ci/monitoring/mixins/grafana_dashboards/dptp.jsonnet (hunk 1 above)
Comment (hongkailiu) [reply cluster/ci/monitoring/mixins/grafana_dashboards/dptp.jsonnet]: I think using namespace is even a bug i made.
We did have some failures ... without namespace ... they do show up.

https://grafana-prow-monitoring-stage.svc.ci.openshift.org/d/8ce131e226b7fd2901c2fce45d4e21c1/dptp-dashboard?orgId=1&fullscreen&panelId=3

Why `namespace` hides my failures? I still cannot explain.
Started reading [doc](https://prometheus.io/docs/prometheus/latest/querying/operators/) ... the 10th time

**Comment by hongkailiu** [comment]
> Could we define a simple JSONNET function so that a user like @bparees could call
> 
> ```jsonnet
> alert_on_job("my-job-name", "my-slack-channel")
> ```
> 
> In order to opt in a channel to job failure messages?

Doable but would be challenging.

For now, we have 2 channels as targets for alerts: 1. our ops and 2. build-cop

This is achieved by 2 parts:
1. prometheus does labels on the alerts (this part is easy)
2. alert-manager checks the labels and send them accordingly (see its [config](https://alertmanager-prow-monitoring.svc.ci.openshift.org/#/status))

For now the config of alert-manager is not controlled by jsonnet (hard-coded) ... Ideally, when we generate its config ... we need to collect the labels/channels used by the alerts ... and generates the config accordingly. I will give it a try tomorrow.

File: cluster/ci/monitoring/mixins/grafana_dashboards/dptp.jsonnet (hunk 1 above)
Comment (hongkailiu) [reply cluster/ci/monitoring/mixins/grafana_dashboards/dptp.jsonnet]: Haha ... I think i know why now ...
periodic jobs have no namespace as a label ... does it sound correct?
I feel odd though :) Will discuss with you tomorrow morning @stevekuznetsov

Comment (hongkailiu) [reply cluster/ci/monitoring/mixins/grafana_dashboards/dptp.jsonnet]: Actually it is the problem of `prowjob_state_transitions`, not `prow_job_labels`.

This should fix it.

**Comment by hongkailiu** [comment]
/hold

Give `alert_on_job("my-job-name", "my-slack-channel")` a try.

**Comment by hongkailiu** [comment]
`alert_on_job("my-job-name", "my-slack-channel")` will be another PR. I would like to test this as soon as possible. The fix at upstream on namespace is independent of this PR (and we do not want to use the `namespace` anyway).

@stevekuznetsov ready to merge if you are OK with the current alert and the panel.

/hold cancel

**Pull request merged**